lib/src/ripple_carry_adder.dart

```diff
@@ -32,5 +38,35 @@ class RippleCarryAdder extends Adder {
     sumList.add(carry);
 
     sum <= sumList.rswizzle();
+    carryOut <= sum[sum.width - 1];
+    out <= sum.slice(sum.width - 2, 0);
+  }
+}
+
+/// A ripple carry adder with a carry in
+class RippleCarryAdder2 extends Adder {
```

Comment:
why does this need to be a separate class? why not make `carry` an optional named argument?